The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/11411453596/job/31755716157) [(plain)](https://github.com/rust-lang/rust/commit/1f2dd7253b453d5be60de79d30db2cb029d28d96/checks/31755716157/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.773 Building wheels for collected packages: reuse
#13 2.774   Building wheel for reuse (pyproject.toml): started
#13 3.023   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.024   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 3.024   Stored in directory: /tmp/pip-ephem-wheel-cache-s9grks02/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.027 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.428 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.428 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 3.958 Collecting virtualenv
#13 3.958 Collecting virtualenv
#13 3.999   Downloading virtualenv-20.27.0-py3-none-any.whl (3.1 MB)
#13 4.063      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.1/3.1 MB 50.0 MB/s eta 0:00:00
#13 4.120 Collecting filelock<4,>=3.12.2
#13 4.125   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.143 Collecting distlib<1,>=0.3.7
#13 4.147   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#13 4.154      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 82.0 MB/s eta 0:00:00
#13 4.187 Collecting platformdirs<5,>=3.9.1
#13 4.191   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.275 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.453 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.27.0
#13 DONE 4.5s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      210880 kB
DirectMap2M:     7129088 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
WARNING: `rust.download-rustc` is enabled. The `rust.channel` option will be overridden by the CI rustc's channel.
downloading https://static.rust-lang.org/dist/2024-10-16/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
 use std::path::{Path, PathBuf};
 use std::{fmt, str};
 
-
 pub(crate) struct StaticFile {
     pub(crate) filename: PathBuf,
     pub(crate) bytes: &'static [u8],
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/src/librustdoc/html/toc/tests.rs" "/checkout/tests/codegen/naked-fn/naked-functions.rs" "/checkout/tests/codegen/optimize-attr-1.rs" "/checkout/tests/codegen/lifetime_start_end.rs" "/checkout/src/librustdoc/html/static_files.rs" "/checkout/tests/codegen/slice-windows-no-bounds-check.rs" "/checkout/src/librustdoc/html/highlight.rs" "/checkout/tests/codegen/align-byval-vector.rs" "/checkout/src/librustdoc/html/mod.rs" "/checkout/src/librustdoc/html/toc.rs" "/checkout/src/librustdoc/html/render/ordered_json.rs" "/checkout/src/librustdoc/html/render/sorted_template/tests.rs" "/checkout/src/librustdoc/html/render/span_map.rs" "/checkout/src/librustdoc/html/render/context.rs" "/checkout/src/librustdoc/html/render/ordered_json/tests.rs" "/checkout/src/librustdoc/html/render/sorted_template.rs" "/checkout/src/librustdoc/html/render/print_item.rs" "/checkout/src/librustdoc/html/render/sidebar.rs" "/checkout/src/librustdoc/html/render/type_layout.rs" "/checkout/src/librustdoc/html/render/write_shared.rs" "/checkout/src/librustdoc/html/render/tests.rs" "/checkout/tests/codegen/intrinsics/typed_swap.rs" "/checkout/tests/codegen/intrinsics/const_eval_select.rs" "/checkout/tests/codegen/intrinsics/aggregate-thin-pointer.rs" "/checkout/tests/codegen/intrinsics/rustc_intrinsic_must_be_overridden.rs" "/checkout/tests/codegen/intrinsics/offset_from.rs" "/checkout/tests/codegen/intrinsics/volatile.rs" "/checkout/tests/codegen/intrinsics/volatile_order.rs" "/checkout/tests/codegen/intrinsics/likely.rs" "/checkout/tests/codegen/intrinsics/three_way_compare.rs" "/checkout/tests/codegen/intrinsics/unchecked_math.rs" "/checkout/tests/codegen/intrinsics/select_unpredictable.rs" "/checkout/tests/codegen/intrinsics/nearby.rs" "/checkout/tests/codegen/intrinsics/transmute.rs" "/checkout/tests/codegen/intrinsics/mask.rs" "/checkout/tests/codegen/intrinsics/ctlz.rs" "/checkout/tests/codegen/intrinsics/nontemporal.rs" "/checkout/tests/codegen/intrinsics/prefetch.rs" "/checkout/tests/codegen/intrinsics/ptr_metadata.rs" "/checkout/tests/codegen/intrinsics/ctpop.rs" "/checkout/tests/codegen/intrinsics/compare_bytes.rs" "/checkout/tests/codegen/intrinsics/offset.rs" "/checkout/tests/codegen/intrinsics/rotate_left.rs" "/checkout/tests/codegen/intrinsics/exact_div.rs" "/checkout/tests/codegen/intrinsics/transmute-x64.rs" "/checkout/tests/codegen/intrinsics/transmute-niched.rs" "/checkout/tests/codegen/array-codegen.rs" "/checkout/src/librustdoc/html/render/search_index/encode.rs" "/checkout/src/librustdoc/html/render/search_index.rs" "/checkout/src/librustdoc/html/render/mod.rs" "/checkout/src/librustdoc/html/render/write_shared/tests.rs" "/checkout/src/librustdoc/html/sources.rs" "/checkout/src/librustdoc/lint.rs" "/checkout/src/librustdoc/visit_lib.rs" "/checkout/src/librustdoc/lib.rs" "/checkout/src/librustdoc/doctest/markdown.rs" "/checkout/src/librustdoc/doctest/runner.rs" "/checkout/src/librustdoc/doctest/rust.rs" "/checkout/src/librustdoc/doctest/make.rs" "/checkout/src/librustdoc/doctest/tests.rs" "/checkout/tests/codegen/sanitizer/scs-attr-check.rs" "/checkout/tests/codegen/sanitizer/dataflow-instrument-functions.rs" "/checkout/src/librustdoc/formats/cache.rs" "/checkout/tests/codegen/naked-fn/naked-noinline.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Fri Oct 18 21:50:46 UTC 2024
  network time: Fri, 18 Oct 2024 21:50:46 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/librustdoc/Cargo.toml
+++ b/src/librustdoc/Cargo.toml
@@ -24,6 +24,7 @@ tracing = "0.1"
 tracing-tree = "0.3.0"
 threadpool = "1.8.1"
 unicode-segmentation = "1.9"
+sha2 = "0.10.8"
 
 [dependencies.tracing-subscriber]
 version = "0.3.3"

--- a/src/librustdoc/html/static/css/rustdoc.css
+++ b/src/librustdoc/html/static/css/rustdoc.css
@@ -44,7 +44,7 @@ xmlns="http://www.w3.org/2000/svg" fill="black" height="18px">\
 	font-style: normal;
 	font-weight: 400;
 	src: local('Fira Sans'),
-		url("FiraSans-Regular-018c141bf0843ffd.woff2") format("woff2");
+		url("FiraSans-Regular-0fe48ade.woff2") format("woff2");
 	font-display: swap;
 }
 @font-face {
@@ -52,7 +52,7 @@ xmlns="http://www.w3.org/2000/svg" fill="black" height="18px">\
 	font-style: normal;
 	font-weight: 500;
 	src: local('Fira Sans Medium'),
-		url("FiraSans-Medium-8f9a781e4970d388.woff2") format("woff2");
+		url("FiraSans-Medium-e1aa3f0a.woff2") format("woff2");
 	font-display: swap;
 }
 
@@ -62,7 +62,7 @@ xmlns="http://www.w3.org/2000/svg" fill="black" height="18px">\
 	font-style: normal;
 	font-weight: 400;
 	src: local('Source Serif 4'),
-		url("SourceSerif4-Regular-46f98efaafac5295.ttf.woff2") format("woff2");
+		url("SourceSerif4-Regular-6b053e98.ttf.woff2") format("woff2");
 	font-display: swap;
 }
 @font-face {
@@ -70,7 +70,7 @@ xmlns="http://www.w3.org/2000/svg" fill="black" height="18px">\
 	font-style: italic;
 	font-weight: 400;
 	src: local('Source Serif 4 Italic'),
-		url("SourceSerif4-It-acdfaf1a8af734b1.ttf.woff2") format("woff2");
+		url("SourceSerif4-It-ca3b17ed.ttf.woff2") format("woff2");
 	font-display: swap;
 }
 @font-face {
@@ -78,7 +78,7 @@ xmlns="http://www.w3.org/2000/svg" fill="black" height="18px">\
 	font-style: normal;
 	font-weight: 700;
 	src: local('Source Serif 4 Bold'),
-		url("SourceSerif4-Bold-a2c9cd1067f8b328.ttf.woff2") format("woff2");
+		url("SourceSerif4-Bold-6d4fd4c0.ttf.woff2") format("woff2");
 	font-display: swap;
 }
 
@@ -89,28 +89,28 @@ xmlns="http://www.w3.org/2000/svg" fill="black" height="18px">\
 	font-weight: 400;
 	/* Avoid using locally installed font because bad versions are in circulation:
 	 * see https://github.com/rust-lang/rust/issues/24355 */
-	src: url("SourceCodePro-Regular-562dcc5011b6de7d.ttf.woff2") format("woff2");
+	src: url("SourceCodePro-Regular-8badfe75.ttf.woff2") format("woff2");
 	font-display: swap;
 }
 @font-face {
 	font-family: 'Source Code Pro';
 	font-style: italic;
 	font-weight: 400;
-	src: url("SourceCodePro-It-1cc31594bf4f1f79.ttf.woff2") format("woff2");
+	src: url("SourceCodePro-It-fc8b9304.ttf.woff2") format("woff2");
 	font-display: swap;
 }
 @font-face {
 	font-family: 'Source Code Pro';
 	font-style: normal;
 	font-weight: 600;
-	src: url("SourceCodePro-Semibold-d899c5a5c4aeb14a.ttf.woff2") format("woff2");
+	src: url("SourceCodePro-Semibold-aa29a496.ttf.woff2") format("woff2");
 	font-display: swap;
 }
 
 /* Avoid using legacy CJK serif fonts in Windows like Batang. */
 @font-face {
 	font-family: 'NanumBarunGothic';
-	src: url("NanumBarunGothic-0f09457c7a19b7c6.ttf.woff2") format("woff2");
+	src: url("NanumBarunGothic-13b3dcba.ttf.woff2") format("woff2");
 	font-display: swap;
 	unicode-range: U+AC00-D7AF, U+1100-11FF, U+3130-318F, U+A960-A97F, U+D7B0-D7FF;
 }

--- a/src/librustdoc/html/static_files.rs
+++ b/src/librustdoc/html/static_files.rs
@@ -3,11 +3,8 @@
 //! All the static files are included here for centralized access in case anything other than the
 //! HTML rendering code (say, the theme checker) needs to access one of these files.
 
-use std::hash::Hasher;
 use std::path::{Path, PathBuf};
 use std::{fmt, str};
-
-use rustc_data_structures::fx::FxHasher;
 
 pub(crate) struct StaticFile {
     pub(crate) filename: PathBuf,
@@ -64,9 +61,11 @@ pub(crate) fn static_filename(filename: &str, contents: &[u8]) -> PathBuf {
 }
 
 fn static_suffix(bytes: &[u8]) -> String {
-    let mut hasher = FxHasher::default();
-    hasher.write(bytes);
-    format!("-{:016x}", hasher.finish())
+    use sha2::Digest;
+    let bytes = sha2::Sha256::digest(bytes);
+    let mut digest = format!("-{bytes:x}");
+    digest.truncate(9);
+    digest
 }
 
 macro_rules! static_files {

